src/sequence_batch_scheduler/sequence_batch_scheduler.cc

```diff
@@ -884,6 +884,29 @@ SequenceBatchScheduler::ReleaseSequenceSlot(
 {
   std::unique_lock<std::mutex> lock(mu_);
 
+  // If we are releasing the slot for a cancelled sequence,
+  // we have to clean up the sequence
+  // otherwise the reaper will try to clean it up again.
+  if (!requests->empty() && requests->front()->IsCancelled()) {
+    // clean-up the cancelled sequences
+    // Normal sequences will already be
+    const InferenceRequest::SequenceId& old_correlation_id =
+        requests->front()->CorrelationId();
+    LOG_VERBOSE(1) << "Releasing canceled "
+                      "CORRID "
+                   << old_correlation_id;
+
+    if (sequence_to_batcherseqslot_map_.find(old_correlation_id) !=
+        sequence_to_batcherseqslot_map_.end()) {
+      sequence_to_batcherseqslot_map_.erase(old_correlation_id);
+    }
+    // remove this correlation ID from expiration checks
+    if (correlation_id_timestamps_.find(old_correlation_id) !=
+        correlation_id_timestamps_.end()) {
+      correlation_id_timestamps_.end();
```

Comment:
should this be `correlation_id_timestamps_.erase();`

Reply:
yes 100%